controller_manager:
  ros__parameters:
    update_rate: 1000  # Update rate in Hz
    controller_manager_timeout: 5.0

    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster
    
    left_hand_controller:
      type: joint_trajectory_controller/JointTrajectoryController

    right_hand_controller:
      type: joint_trajectory_controller/JointTrajectoryController

left_hand_controller:
  ros__parameters:
    allow_partial_joints_goal: True
    action_monitor_rate: 20.0
    open_loop_control: true
    # constraints:
    #   stopped_velocity_tolerance: 0.01
    #   goal_time: 0.0
    #   left_hand_stepper_joint:
    #     trajectory: 0.05
    #     goal: 0.03
    joints:
      - left_hand_stepper_joint
      - left_hand_thumb_solenoid_joint
      - left_hand_thumb_servo_joint
      - left_hand_index_finger_solenoid_joint
      - left_hand_index_finger_servo_joint
      - left_hand_middle_finger_solenoid_joint
      - left_hand_middle_finger_servo_joint
      - left_hand_ring_finger_solenoid_joint
      - left_hand_ring_finger_servo_joint
      - left_hand_pinky_solenoid_joint
      - left_hand_pinky_servo_joint
    gains:
      left_hand_stepper_joint: {p: 1.0, i: 0.0, d: 1.0}
    command_interfaces:
      - position
    state_interfaces:
      - position

# right_hand_controller:
#   ros__parameters:
#     allow_partial_joints_goal: True
#     action_monitor_rate: 20.0
#     open_loop_control: true
#     # constraints:
#     #   stopped_velocity_tolerance: 0.01
#     #   goal_time: 0.0
#     #   right_hand_stepper_joint:
#     #     trajectory: 0.05
#     #     goal: 0.03
#     joints:
#       - right_hand_stepper_joint
#       - right_hand_thumb_solenoid_joint
#       - right_hand_thumb_servo_joint
#       - right_hand_index_finger_solenoid_joint
#       - right_hand_index_finger_servo_joint
#       - right_hand_middle_finger_solenoid_joint
#       - right_hand_middle_finger_servo_joint
#       - right_hand_ring_finger_solenoid_joint
#       - right_hand_ring_finger_servo_joint
#       - right_hand_pinky_solenoid_joint
#       - right_hand_pinky_servo_joint
#     gains:
#       right_hand_stepper_joint: {p: 1.0, i: 0.0, d: 1.0}
#     command_interfaces:
#       - position
#     state_interfaces:
#       - position
